neo4j-to-neptune/src/main/java/com/amazonaws/services/neptune/util/GzipCompressUtils.java

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+A copy of the License is located at
+    http://www.apache.org/licenses/LICENSE-2.0
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+*/
+
+package com.amazonaws.services.neptune.util;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for gzip compression of CSV files
+ */
+public class GzipCompressUtils {
+    /**
+     * Upload Neptune vertices and edges CSV files asynchronously
+     */
+    public static void compressCsvFiles(String uploadDirectoryPath, Boolean isCompressDelete) throws Exception {
```

Comment:
Nit: This utility class does not have any awareness of how the csv files came to be at the given location so would be appropriate to remove the reference to 'upload' in the param name.

If you don't expect to handle null value, should use primitive instead.

```suggestion
    public static void compressCsvFiles(String directoryPath, boolean isCompressDelete) throws Exception {
```